test/Api.Test/Tools/Validators/SendRotationValidatorTests.cs

```diff
@@ -0,0 +1,130 @@
+﻿using Bit.Api.Tools.Models;
+using Bit.Api.Tools.Models.Request;
+using Bit.Api.Tools.Validators;
+using Bit.Core.Entities;
+using Bit.Core.Exceptions;
+using Bit.Core.Tools.Entities;
+using Bit.Core.Tools.Enums;
+using Bit.Core.Tools.Models.Data;
+using Bit.Core.Tools.Repositories;
+using Bit.Core.Tools.Services;
+using Bit.Test.Common.AutoFixture.Attributes;
+using NSubstitute;
+using Xunit;
+
+namespace Bit.Api.Test.Tools.Validators;
+
+[SutProviderCustomize]
+public class SendRotationValidatorTests
+{
+    [Fact]
+    public async Task ValidateAsync_Success()
+    {
+        // Arrange
+        var sendService = Substitute.For<ISendService>();
+        var sendRepository = Substitute.For<ISendRepository>();
+
+        var sut = new SendRotationValidator(
+            sendService,
+            sendRepository
+        );
+
+        var user = new User { Id = new Guid() };
+        var sends = CreateSends();
+        var userSends = sends.Select(s =>
+            new Send { Id = s.Id.Value, Type = s.Type, Data = "{}" }).ToList();
+        sendRepository.GetManyByUserIdAsync(user.Id).Returns(userSends);
+
+        // Act
+        var result = await sut.ValidateAsync(user, sends);
+
+        // Assert
+        Assert.Equal(userSends, result);
+    }
+
+    [Fact]
+    public async Task ValidateAsync_InputMissingUserSend_Throws()
+    {
+        // Arrange
+        var sendService = Substitute.For<ISendService>();
+        var sendRepository = Substitute.For<ISendRepository>();
+
+        var sut = new SendRotationValidator(
+            sendService,
+            sendRepository
+        );
+
+        var user = new User { Id = new Guid() };
+        var sends = CreateSends();
+        var userSends = sends.Select(s =>
+            new Send { Id = s.Id.Value, Type = s.Type, Data = "{}" }).ToList();
+        userSends.Add(new Send { Id = Guid.NewGuid(), Data = "{}" });
+        sendRepository.GetManyByUserIdAsync(user.Id).Returns(userSends);
+
+        // Act, Assert
+        await Assert.ThrowsAsync<BadRequestException>(async () => await sut.ValidateAsync(user, sends));
+    }
+
+    [Fact]
+    public async Task ValidateAsync_SendDoesNotBelongToUser_SendNotReturned()
+    {
+        // Arrange
+        var sendService = Substitute.For<ISendService>();
+        var sendRepository = Substitute.For<ISendRepository>();
+
+        var sut = new SendRotationValidator(
+            sendService,
+            sendRepository
+        );
+
+        var user = new User { Id = new Guid() };
+        var sends = CreateSends();
+        var userSends = sends.Select(s =>
+            new Send { Id = s.Id.Value, Type = s.Type, Data = "{}" }).ToList();
+        userSends.RemoveAt(0);
```

Comment:
⛏️  This is not checking whether the send belongs to the user, since its userId hasn't been set up to be different than the input userId--in fact, none of the sends have their userid set. Instead, it's checking that `GetManyByUserId` no longer lists the send, which could happen for many reasons--for example, the send expired and was deleted. The name of the test case should be updated to reflect the actual state of the test: that the send is no longer in the system.

⚠️ The mock data should not be derived from the input data. In a best-case scenario, the test will be brittle and fail when the code changes. In a worst-case, it'll flag a false-positive and the validation will allow records it shouldn't. Consider hard-coding the stub's return values instead of deriving them from the input.